phase2/src/load_circuit.rs

```diff
@@ -0,0 +1,38 @@
+use algebra::{CanonicalDeserialize, CanonicalSerialize, PairingEngine, SerializationError};
+use r1cs_core::Matrix;
+use setup_utils::Error;
+use std::io::{Read, Write};
+
+// For serialization of the constraint system
+#[derive(Debug, PartialEq, CanonicalDeserialize, CanonicalSerialize, Clone)]
+pub struct Matrices<E: PairingEngine> {
```

Comment:
is this a copy of an existing struct? if so, can we use that directly?

Reply:
It copies ConstraintMatrices from zexe. ConstraintMatrices does not implement or derive CanonicalSerialize or CanonicalDeserialize, making this the simplest workaround. Afaik the only way to avoid copying the struct is to have the zexe equivalent derive serialization, and update our dependency for it